wallet/src/de/schildbach/wallet/ui/dashpay/FrequentContactsLiveData.kt

```diff
@@ -0,0 +1,113 @@
+package de.schildbach.wallet.ui.dashpay
+
+import android.text.format.DateUtils
+import androidx.lifecycle.LiveData
+import de.schildbach.wallet.WalletApplication
+import de.schildbach.wallet.data.UsernameSearchResult
+import de.schildbach.wallet.data.UsernameSortOrderBy
+import de.schildbach.wallet.livedata.Resource
+import de.schildbach.wallet.livedata.Status
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.dashevo.dashpay.BlockchainIdentity
+import java.util.*
+
+class FrequentContactsLiveData(val walletApplication: WalletApplication, private val platformRepo: PlatformRepo) : LiveData<Resource<List<UsernameSearchResult>>>(), OnContactsUpdated {
+
+    companion object {
+        const val TIMESPAN: Long = DateUtils.DAY_IN_MILLIS * 90 // 90 days
+        const val TOP_CONTACT_COUNT = 4
+    }
+
+    private var listening = false
+
+    override fun onActive() {
+        maybeAddEventListener()
+        onContactsUpdated()
+    }
+
+    override fun onInactive() {
+        maybeRemoveEventListener()
+    }
+
+    private fun maybeAddEventListener() {
+        if (!listening && hasActiveObservers()) {
+            platformRepo.addContactsUpdatedListener(this)
+            listening = true
+        }
+    }
+
+    private fun maybeRemoveEventListener() {
+        if (listening) {
+            platformRepo.removeContactsUpdatedListener(this)
+            listening = false
+        }
+    }
+
+    override fun onContactsUpdated() {
+        getFrequentContacts()
+    }
+
+    fun getFrequentContacts() {
+        GlobalScope.launch(Dispatchers.IO) {
+            val contactRequests = platformRepo.searchContacts("", UsernameSortOrderBy.DATE_ADDED)
+            when (contactRequests.status) {
+                Status.SUCCESS -> {
+                    val blockchainIdentity = platformRepo.getBlockchainIdentity() ?: return@launch
+                    val threeMonthsAgo = Date().time - TIMESPAN
+
```

Comment:
This is where the contactRequests.data should be filtered to only use the proper contacts for the rest of the "calculations"

Reply:
Resolved here: https://github.com/dashevo/dash-wallet/pull/475/commits/1cc68c78496e0a2248c64dccb26bd8b16a617016